crates/transaction-emitter-lib/src/emitter/transaction_executor.rs

```diff
@@ -25,48 +26,94 @@ impl RestApiTransactionExecutor {
         self.rest_clients.choose(&mut rng).unwrap()
     }
 
-    async fn submit_and_check(
+    fn random_rest_client_from_rng<R>(&self, rng: &mut R) -> &RestClient
+    where
+        R: Rng + ?Sized,
+    {
+        self.rest_clients.choose(rng).unwrap()
+    }
+
+    async fn submit_check_and_retry(
         &self,
         txn: &SignedTransaction,
-        failure_counter: &AtomicUsize,
+        failure_counter: &[AtomicUsize],
+        run_seed: u64,
     ) -> Result<()> {
-        let rest_client = self.random_rest_client();
-        if let Err(err) = rest_client.submit_bcs(txn).await {
-            sample!(
-                SampleRate::Duration(Duration::from_secs(60)),
-                warn!(
-                    "[{}] Failed submitting transaction: {}",
-                    rest_client.path_prefix_string(),
-                    err,
-                )
-            );
-            failure_counter.fetch_add(1, std::sync::atomic::Ordering::Relaxed);
-            // even if txn fails submitting, it might get committed, so wait to see if that is the case.
-        }
-        if let Err(err) = rest_client
-            .wait_for_transaction_by_hash(
-                txn.clone().committed_hash(),
-                txn.expiration_timestamp_secs(),
-                None,
-                Some(Duration::from_secs(10)),
+        for i in 0..self.max_retries {
+            // All transactions from the same sender, need to be submitted to the same client
+            // in the same retry round, so that they are not placed in parking lot.
+            // Do so by selecting a client via seeded random selection.
```

Comment:
@msmouse - here, we pick a pseudo random sender, with sender being part of the seed.

Reply:
I see.. I read the summary reversely.. English is hard..